pkg/controller.v1alpha3/experiment/experiment_util.go

```diff
@@ -109,3 +111,24 @@ func (r *ReconcileExperiment) updateFinalizers(instance *experimentsv1alpha3.Exp
 		return reconcile.Result{Requeue: true}, err
 	}
 }
+
+func (r *ReconcileExperiment) terminateSuggestion(instance *experimentsv1alpha3.Experiment) error {
+	suggestion := &suggestionsv1alpha3.Suggestion{}
+	err := r.Get(context.TODO(), types.NamespacedName{
+		Namespace: instance.Namespace,
+		Name:      instance.Name,
+	}, suggestion)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	if suggestion.IsCompleted() {
+		return nil
+	}
+	msg := "Suggestion is succeeded"
+	suggestion.MarkSuggestionStatusSucceeded(suggestionController.SuggestionSucceededReason, msg)
+
+	return r.UpdateSuggestion(suggestion, 0)
```

Comment:
Do you need to update Suggestion with Request Count = 0?
And I believe we didn't use `suggestionRequest` in `UpdateSuggestion` function: https://github.com/kubeflow/katib/blob/master/pkg/controller.v1alpha3/experiment/suggestion/suggestion.go#L82.
If you want to update suggestion you can do it with instance, like here: https://github.com/kubeflow/katib/blob/master/pkg/controller.v1alpha3/experiment/experiment_controller.go#L390

Reply:
@andreyvelich I've changed with `r.Update(context.TODO(), suggestion)`, but it still doesn't work in e2e tests. I have no idea.

Reply:
@sperlingxx I answered bellow.